File: project/lib/core/action/AttackUnitAction.js
utils = require("../utils");
Action = require("./Action")


class AttackUnitAction extends Action{
    constructor(unit, from, to, enemy, kills, damage, weapon, armour){
        super();

        if(unit === undefined){
            throw new utils.Exception("Not defined");
        }

        if(from === undefined){
            throw new utils.Exception("Not defined");
        }

        if(to === undefined){
            throw new utils.Exception("Not defined");
        }

        if(enemy === undefined){
            throw new utils.Exception("Not defined");
        }

        if(kills === undefined){
            throw new utils.Exception("Not defined");
        }

        this.unit = unit;
        this.from = from;
        this.to = to;
        this.enemy = enemy;
        this.kills = kills;
        this.damage = damage;
        this.weapon = weapon;
        this.armour = armour;
        this.type = "Attack unit";
    }
}


module.exports = AttackUnitAction;

File: UnityClient/Assets/src/lib/action/AttackUnitAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Assets.src.lib.entities;


namespace Assets.src.lib.action
{
    public class AttackUnitAction : Action
    {
        public Unit unit { get; set; }
        public Point from { get; set; }
        public Point to { get; set; }
        public bool kills { get; set; }
        public Unit enemy { get; set; }
        public int damage { get; set; }
        public int weapon { get; set; }
        public int armour { get; set; }
    }
}
